src/transformState.js

```diff
@@ -5,7 +5,21 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      for (const data in action.extraData) {
+        state[data] = action.extraData[data];
+      }
+    } else if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete state[key];
+      }
+    } else if (action.type === 'clear') {
+      Object.keys(state).forEach(key => delete state[key]);
+    }
```

Comment:
Use `switch...case` instead of `if...else if...` - it's easier to read and maintain.